syncer/syncer.go

```diff
@@ -931,11 +931,18 @@ func (s *Syncer) syncDDL(tctx *tcontext.Context, queueBucket string, db *DBConn,
 			}
 		}
 		if err != nil {
+			s.jobWg.Done()
+			s.execError.Set(err)
+			if !utils.IsContextCanceledError(err) {
+				err = s.handleEventError(err, &sqlJob.startLocation, &sqlJob.currentLocation)
+				s.runFatalChan <- unit.NewProcessError(err)
+			}
 			s.appendExecErrors(&ExecErrorContext{
 				err:      err,
 				location: sqlJob.currentLocation.Clone(),
 				jobs:     fmt.Sprintf("%v", sqlJob.ddls),
 			})
+			continue
```

Comment:
forward: "can we just rollback checkpoint as we did before?" https://github.com/pingcap/dm/pull/1032#discussion_r489988212

I think that's no need, we could use caller's rollback

Reply:
So no need to change `addjob` in #1032, right?